modules/distributor/distributor.go

```diff
@@ -603,7 +625,28 @@ func requestsByTraceID(batches []*v1.ResourceSpans, userID string, spanCount int
 		traces = append(traces, r)
 	}
 
-	return keys, traces, nil
+	return keys, traces, truncatedAttributeCount, nil
+}
+
+// find and truncate the span attributes that are too large
+func processAttributes(attributes []*v1_common.KeyValue, maxAttrSize int, count *int) {
+	for _, attr := range attributes {
+		if len(attr.Key) > maxAttrSize {
+			attr.Key = attr.Key[:maxAttrSize] + "_truncated"
+			*count++
+		}
+
+		switch value := attr.GetValue().Value.(type) {
+		case *v1_common.AnyValue_StringValue:
+			if len(value.StringValue) > maxAttrSize {
+				fmt.Printf("size: %d", len(value.StringValue))
+				value.StringValue = value.StringValue[:maxAttrSize] + "_truncated"
```

Comment:
this is quite expensive as it's going to alloc a completely new string every time.

sane but not as nice option:
1) just produce a metric that indicates number of truncated attributes

crazy but fun option:
2) rewrite the byte array underlying the string to contain "_truncated" at the right position

Reply:
the benchmark for the scenario where every attribute needs to be truncated, adding the "_truncated" had a ~400+% degradation. And only 66% if we just truncate without adding the "_truncated". I'm in favor of just reporting the metrics without adding "_truncated". Wdyt?

<details><summary> benchmarks</summary>

4 traces, 300 batches and 5000 spans each run

with adding "_truncated"

```
                 │ before.out  │        after_add_truncated.out         │
                 │   sec/op    │   sec/op      vs base                  │
TestsByRequestID   402.3m ± 7%   2384.8m ± 10%  +492.83% (p=0.000 n=10)
```

without adding "_truncated" to truncated attribute values

```
                 │  before.out  │         after-notruncate.out         │
                 │    sec/op    │    sec/op     vs base                │
TestsByRequestID   357.1m ± 52%   594.0m ± 34%  +66.33% (p=0.001 n=10)

                 │  before.out  │      after-notruncate.out      │
                 │     B/op     │     B/op      vs base          │
TestsByRequestID   58.71Mi ± 0%   58.71Mi ± 0%  ~ (p=1.000 n=10)

                 │ before.out  │      after-notruncate.out       │
                 │  allocs/op  │  allocs/op   vs base            │
TestsByRequestID   5.100k ± 0%   5.100k ± 0%  ~ (p=1.000 n=10) ¹
```

</details>